### PR TITLE
[TEST]Add Openai protocol test 

### PR DESCRIPTION
**Pull request by yutianchen666**
add test case like ray-llm project

### CODE REVIEW

**Comment by carsonwang** [comment]
Please also refer to `test_router.py` and add it. Please add the license header and mention the original file it refers. You can check other OpenAI implementation python files for examples.

File: tests/openai/test_openai_protocol.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+os.environ["no_proxy"] = "localhost,127.0.0.1"
+os.environ["OPENAI_API_BASE"] = "http://localhost:8000/v1"
+os.environ["OPENAI_API_KEY"] = "YOUR_OPEN_AI_KEY"
+os.environ["OPENAI_BASE_URL"] = "http://localhost:8000/v1"
+openai_base_url = os.environ["OPENAI_BASE_URL"]
+openai_api_key = os.environ["OPENAI_API_KEY"]
+client = OpenAI(base_url=openai_base_url, api_key=openai_api_key)
+
+
+def start_serve(model_name):
```
Comment (carsonwang) [inline tests/openai/test_openai_protocol.py:16]: The same function also appears in `test_example_simple.py`. Can we put it in a common place?

File: tests/openai/test_openai_protocol.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    assert chat_completion.id
+    assert isinstance(chat_completion.choices, list)
+    assert chat_completion.choices[0].delta == {} or hasattr(
+        chat_completion.choices[0].delta, "content"
+    )
+    assert chat_completion.choices[0].model_fields["finish_reason"]
+    assert i > 4
+
+
+def completions_stream_bad_request(openai_testing_model):  # noqa: F811
```
Comment (carsonwang) [inline tests/openai/test_openai_protocol.py:135]: We can remove 4 methods here starting with `completions_` as they are not used.

File: tests/openai/test_openai_protocol.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+        (model, test_func)
+        for model in ["gpt2"]
+        for test_func in [
+            "models",
+            # "completions",
+            # "completions_stream",
+            "chat",
+            "chat_stream",
+            # "chat_bad_request",
+            # "chat_stream_bad_request"
```
Comment (carsonwang) [inline tests/openai/test_openai_protocol.py:175]: Can we enable these two starting with `chat_` and remove others starting with `completions_` ?

**Review by carsonwang — COMMENTED** [review]
Please merge the latest main branch to remove the application code in this PR. The lint issues in application have been fixed.

File: tests/openai/test_openai_protocol.py (hunk 1 above)
Comment (yutianchen666) [reply tests/openai/test_openai_protocol.py:16]: ok,i will reorganize the function locations

File: tests/openai/test_openai_protocol.py (hunk 2 above)
Comment (yutianchen666) [reply tests/openai/test_openai_protocol.py:135]: ok

File: tests/openai/test_openai_protocol.py (hunk 3 above)
Comment (yutianchen666) [reply tests/openai/test_openai_protocol.py:175]: I'm trying make them enable

**Review by carsonwang — APPROVED** [review]

**Pull request merged**